spec.html

```diff
@@ -84,32 +84,28 @@ <h1>Runtime Semantics: PerformPromiseAny ( _iteratorRecord_, _constructor_, _res
       1. Let _remainingElementsCount_ be a new Record { [[Value]]: 1 }.
       1. Let _index_ be 0.
       1. Let _promiseResolve_ be ? Get(_constructor_, `"resolve"`).
-      1. If IsCallable(_promiseResolve_) is *false*, throw a *TypeError* exception.
+      1. If ! IsCallable(_promiseResolve_) is *false*, throw a *TypeError* exception.
       1. Repeat,
         1. Let _next_ be IteratorStep(_iteratorRecord_).
         1. If _next_ is an abrupt completion, set _iteratorRecord_.[[Done]] to *true*.
         1. ReturnIfAbrupt(_next_).
         1. If _next_ is *false*, then
           1. Set _iteratorRecord_.[[Done]] to *true*.
-          1. Set _remainingElementsCount_.[[Value]] to _remainingElementsCount_.[[Value]] - 1.
-          1. If _remainingElementsCount_.[[Value]] is 0, then
-            1. Let _valuesArray_ be CreateArrayFromList(_values_).
-            1. Perform ? Call(_resultCapability_.[[Resolve]], *undefined*, &laquo; _valuesArray_ &raquo;).
           1. Return _resultCapability_.[[Promise]].
         1. Let _nextValue_ be IteratorValue(_next_).
         1. If _nextValue_ is an abrupt completion, set _iteratorRecord_.[[Done]] to *true*.
         1. ReturnIfAbrupt(_nextValue_).
         1. Append *undefined* to _values_.
         1. Let _nextPromise_ be ? Call(_promiseResolve_, _constructor_, &laquo; _nextValue_ &raquo;).
         1. Let _steps_ be the algorithm steps defined in <emu-xref href="#sec-promise.any-reject-element-functions" title></emu-xref>.
-        1. Let _rejectElement_ be CreateBuiltinFunction(_steps_, &laquo; [[AlreadyCalled]], [[Index]], [[Values]], [[Capability]], [[RemainingElements]] &raquo;).
+        1. Let _rejectElement_ be ! CreateBuiltinFunction(_steps_, &laquo; [[AlreadyCalled]], [[Index]], [[Values]], [[Capability]], [[RemainingElements]] &raquo;).
```

Comment:
Why `!`?

Reply:
This matches changes in https://github.com/tc39/proposal-promise-allSettled/commit/7239fa6c75241f437941dabf7babeaf8dff95a39#diff-3540caefa502006d8a33cb1385720803R67. Although `Promise.all` doesn't have `!` in current spec text https://tc39.es/ecma262/#sec-performpromiseall

Reply:
IIUC `Promise.all` should also be preceeded by `!`. Should I create PR in ecma262 repo?

Reply:
In this particular case, IIUC `CreateBuiltinFunction` can in fact return an abrupt completion (because `steps` comes from `Promise.any` Reject Element Functions which has a few `?`s). No?

`Promise.all` seems like a good candidate, though.

Reply:
oh indeed

Reply:
I changed `?` back to `!` per discussion in https://github.com/tc39/ecma262/pull/1607